### ISSUE TITLE
508-defect-3 [SCREENREADER]: VAOS - Loading spinner for checking if you can make an appointment is not announced

### ISSUE THREAD

**Issue by noahgelman** [department-of-veterans-affairs/va.gov-team#18697]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Noah Gelman

## User Story or Problem Statement

"Not seeing the type of care you need?" heading and related content box is not reachable when navigating forward by keyboard, only when navigating backwards.

https://staging.va.gov/health-care/schedule-view-va-appointments/appointments/new-appointment/preferred-date

## Details

When picking a date and the app checks if you can make an appointment. The loading spinner that pops up is not announced to a user. The page needs to announce to the user that something is happening.

## Acceptance Criteria

- [ ] Add role="status" to the loading spinner.

**Comment by jbalboni** [comment]
Hi @noahgelman, we're using the `LoadingIndicator` component from the design system for these spinners, which have a role of `progressbar` already. I think if that needs to be changed, it would be your team and/or the design system team making that change.

**Comment by noahgelman** [comment]
@jbalboni Read the following page on usage for progress bar to make sure you have all the related aria labels added: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_progressbar_role

**Comment by jbalboni** [comment]
@noahgelman Sorry, does that comment mean that the `progressbar` role is fine for this component, but there are some missing attributes?

I'm trying to clarify this because I can probably have our team make a simple change to add the `valuemin` and `valuemax` aria attributes, but if there are broader questions about the `role` being used, I can't really make that decision for a global component used by lots of other apps.

**Comment by jbalboni** [comment]
Based on the conversation in [this Slack thread](https://dsva.slack.com/archives/C8E985R32/p1612198438025300), it appears that I should do some focus handling around the loading indicator on this page, and leave changes to the indicator itself to the VSP design system team.

**Comment by noahgelman** [comment]
@jbalboni 

Sorry for the delay in response.

Loading spinners should have have a progress bar role. Those are for static items.

Use the role="status" instead and include some text inside that says "Loading" which can be hidden.

**Comment by jbalboni** [comment]
@noahgelman Changes for this should be up on staging. Since the role issue isn't resolved for the LoadingSpinner component, I just set focus to the loading spinner when it's visible and set focus back to an appropriate spot after loading is finished.
